### ISSUE TITLE
Const generics: Syntax Error: expected type

### ISSUE THREAD

**Issue by callym** [rust-analyzer/rust-analyzer#7232]
So the following code gives errors at the `let s: S<true|false> = S;` and the `fn only_true(s: S<true>)`, saying `Syntax Error: expected type`

When I run the code with `cargo run`, it all works as expected - (the second `only_true` call gives a compiler error ``expected `true`, found `false` ``, commenting that out prints out the following, as it should do.

```
true
false
only true
``` 

Cargo version: `cargo 1.50.0-nightly (75d5d8cff 2020-12-22)`
Rust version: `rustc 1.51.0-nightly (fde692739 2021-01-02)`
RA version: `0.3.445-nightly`

```rust
#[derive(Copy, Clone)]
struct S<const B: bool>;

impl<const B: bool> S<B> {
    fn get_b(&self) -> bool {
        B
    }
}

fn main() {
    let s_true: S<true> = S;
    let s_false: S<false> = S;

    use_s(s_true);
    use_s(s_false);

    only_true(s_true);
    only_true(s_false);
}

fn use_s<const B: bool>(s: S<B>) {
    println!("{}", s.get_b());
}

fn only_true(s: S<true>) {
    println!("only true");
}
```

I've actually been sort of keen to get into working on RA/Rust/etc issues, so if this looks like it would be a good intro issue, I'd love to give a go at fixing it if there's any pointers. 

<details>
  <summary>VSCode output</summary>
  
```
  INFO [10/01/2021, 11:38:54]: Extension version: 0.3.445-nightly
INFO [10/01/2021, 11:38:54]: Using configuration {
  cargoRunner: null,
  runnableEnv: null,
  inlayHints: {
    enable: true,
    chainingHints: true,
    maxLength: null,
    parameterHints: true,
    typeHints: true
  },
  updates: { channel: 'nightly', askBeforeDownload: false },
  server: { path: null, extraEnv: null },
  trace: { server: 'off', extension: false },
  debug: {
    engine: 'auto',
    sourceFileMap: {
      '/rustc/<id>': '${env:USERPROFILE}/.rustup/toolchains/<toolchain-id>/lib/rustlib/src/rust'
    },
    openDebugPane: false,
    engineSettings: {}
  },
  assist: { importMergeBehavior: 'full', importPrefix: 'plain' },
  callInfo: { full: true },
  cargo: {
    autoreload: true,
    allFeatures: false,
    features: [],
    loadOutDirsFromCheck: true,
    noDefaultFeatures: false,
    target: null,
    noSysroot: false
  },
  checkOnSave: {
    enable: true,
    allFeatures: null,
    allTargets: true,
    command: 'clippy',
    noDefaultFeatures: null,
    target: null,
    extraArgs: [],
    features: null,
    overrideCommand: null
  },
  completion: {
    addCallArgumentSnippets: true,
    addCallParenthesis: true,
    postfix: { enable: true },
    autoimport: { enable: true }
  },
  diagnostics: {
    enable: true,
    enableExperimental: true,
    disabled: [],
    warningsAsHint: [],
    warningsAsInfo: []
  },
  files: { watcher: 'client' },
  hoverActions: {
    debug: true,
    enable: true,
    gotoTypeDef: true,
    implementations: true,
    run: true,
    linksInHover: true
  },
  lens: {
    debug: true,
    enable: true,
    implementations: true,
    run: true,
    methodReferences: true
  },
  linkedProjects: [],
  lruCapacity: null,
  notifications: { cargoTomlNotFound: true },
  procMacro: { enable: true, server: null },
  runnables: { overrideCargo: null, cargoExtraArgs: [] },
  rustcSource: null,
  rustfmt: { extraArgs: [], overrideCommand: null }
}
INFO [10/01/2021, 11:38:54]: PersistentState: {
  lastCheck: 1610106232376,
  releaseId: 36122250,
  serverVersion: '0.3.445-nightly'
}
INFO [10/01/2021, 11:38:54]: Using server binary at /home/callym/.config/Code/User/globalStorage/matklad.rust-analyzer/rust-analyzer-x86_64-unknown-linux-gnu
```
</details>

**Comment by Veykril** [comment]
The fix for this is rather simple, in https://github.com/rust-analyzer/rust-analyzer/blob/dcbb77cb6c75db9b300abe0fe6ed4668e2ef1193/crates/parser/src/grammar/type_args.rs#L30-L63
we only check with `is_literal` for literals which doesn't cover the `true` and `false` keywords. So adding a branch for those two tokens here should be good.
Feel free to work on this and ask for more info if needed 👍
